contracts/dao/USDTLpListaDistributor.sol

```diff
@@ -0,0 +1,449 @@
+    IERC20(lpToken).safeTransfer(distributor, amount);
+    emit WithdrawLp(address(usdt), distributor, distributor, amount);
+
+    emit LPTokenWithdrawn(address(lpToken), _account, amount);
+  }
+
+  // Claim CAKE reward; perform by staking vault or user
+  function _claimStakingReward(address _account) internal returns (uint256) {
+    _updateStakeReward(_account, lpBalanceOf[_account], lpTotalSupply);
+    uint256 amount = stakeStoredPendingReward[_account];
+    delete stakeStoredPendingReward[_account];
+
+    emit StakeRewardClaimed(_account, amount);
+    return amount;
+  }
+
+  // Calculate CAKE reward if any write operation is performed
+  function _updateStakeReward(address _account, uint256 balance, uint256 supply) internal {
+    // update reward
+    uint256 updated = stakePeriodFinish;
+    if (updated > block.timestamp) updated = block.timestamp;
+    uint256 duration = updated - stakeLastUpdate;
+    if (duration > 0) stakeLastUpdate = uint32(updated);
+
+    if (duration > 0 && supply > 0) {
+      stakeRewardIntegral += (duration * stakeRewardRate * 1e18) / supply;
+    }
+    if (_account != address(0)) {
+      uint256 integralFor = stakeRewardIntegralFor[_account];
+      if (stakeRewardIntegral > integralFor) {
+        stakeStoredPendingReward[_account] += (balance * (stakeRewardIntegral - integralFor)) / 1e18;
+        stakeRewardIntegralFor[_account] = stakeRewardIntegral;
+      }
+    }
+  }
+
+  /* ==================== Role-Based Functions ==================== */
+
+  /**
+   * @dev notify staking reward, only staking vault can call this function
+   * @param amount reward amount
+   */
+  function notifyStakingReward(uint256 amount) external onlyStakeVault {
+    _updateStakeReward(address(0), 0, lpTotalSupply);
+    uint256 _periodFinish = stakePeriodFinish;
+    if (block.timestamp < _periodFinish) {
+      uint256 remaining = _periodFinish - block.timestamp;
+      amount += remaining * stakeRewardRate;
+    }
+
+    stakeRewardRate = amount / REWARD_DURATION;
+
+    stakeLastUpdate = block.timestamp;
+    stakePeriodFinish = block.timestamp + REWARD_DURATION;
+  }
+
+  /**
+   * @dev claim reward, only staking vault can call this function
+   * @param _account account address
+   * @return reward amount
+   */
+  function vaultClaimStakingReward(address _account) external onlyStakeVault returns (uint256) {
+    return _claimStakingReward(_account);
+  }
+
+  /**
+   * @dev set stake vault address
+   * @param _stakeVault stake vault address
+   */
+  function setStakeVault(address _stakeVault) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    require(_stakeVault != address(0), "stake vault is the zero address");
+    stakeVault = _stakeVault;
+  }
+
+  /**
+   * @dev emergency withdraw LP token from farming contract
+   */
+  function emergencyWithdraw() external onlyRole(DEFAULT_ADMIN_ROLE) notInEmergencyMode {
+    uint256 lpAmount = IERC20(lpToken).balanceOf(address(this));
```

Comment:
support disable emergencyMode and stake all LP balance to farming contract again.